xbmc/epg/GUIEPGGridContainer.cpp

```diff
@@ -1429,9 +1429,11 @@ bool CGUIEPGGridContainer::OnMouseWheel(char wheel, const CPoint &point)
 
 CPVRChannelPtr CGUIEPGGridContainer::GetChannel(int iIndex)
 {
+  CSingleLock lock(m_critSection);
+
   if (iIndex >= 0 && (size_t) iIndex < m_channelItems.size())
   {
-    CFileItemPtr fileItem = m_channelItems[iIndex];
+    const CFileItemPtr &fileItem = m_channelItems[iIndex];
```

Comment:
you're conflating references with reference counted std::shared_ptr, they don't work the same way. std::shared_ptr increases the ref count in the copy constructor, just taking a regular reference to a heap based object gives you no guarantee that the object will still be valid a second later.